src/cache.ts

```diff
@@ -0,0 +1,32 @@
+import LruCache from "lru-cache";
+
+// Creates a function that memoizes its result using an LRU cache.
+//
+// The cache key is a string created from the arguments using
+// `makeKey`.
+export function memoizeLru<Args extends unknown[], V>(
+  f: (...args: Args) => Promise<V>,
+  makeKey: (...args: Args) => string,
+  options?: LruCache.Options<string, { value: V }>
+): (...args: Args) => Promise<V> {
+  const cache = new LruCache(options || { max: 500 });
+
+  if (! window.cache) {
+    window.cache = [cache];
+  } else {
+    window.cache.push(cache);
+  }
```

Comment:
We should not leave this in here, it was meant only for debugging. Otherwise these objects will take up memory.